packages/slice-machine/lib/mock/Slice.ts

```diff
@@ -1,53 +1,80 @@
-import { FieldsSM } from "@slicemachine/core/build/models/Fields";
-import { SliceSM, VariationSM } from "@slicemachine/core/build/models/Slice";
+import {
+  SharedSlice,
+  SlicesTypes,
+  Variation,
+} from "@prismicio/types-internal/lib/customtypes/widgets/slices";
+import {
+  SharedSliceMockConfig,
+  VariationMockConfig,
+  generateSliceMock,
+  renderSliceMock,
+} from "@prismicio/mocks";
+import { buildFieldsMockConfig } from "./LegacyMockConfig";
+import { Slices, SliceSM } from "@slicemachine/core/build/models/Slice";
 
-const createEmptyMock = (
-  sliceName: string,
-  variation: VariationSM
-): Models.VariationMock => ({
-  variation: variation.id,
-  name: variation.name,
-  slice_type: snakelize(sliceName),
-  items: [],
-  primary: {},
-});
-
-export default function MockSlice(
-  sliceName: string,
-  model: SliceSM,
-  mockConfig: Record<string, Record<string, Record<string, unknown>>> // not sure about this one.
-): Models.SliceMock {
-  const variations = model.variations.map((variation) => {
-    const mock = createEmptyMock(sliceName, variation);
-    const handler: (
-      fields?: FieldsSM,
-      mocks?: Record<string, unknown>
-    ) => Models.VariationMock["primary"] = handleFields(Widgets);
-
-    const maybeMockConfig = mockConfig?.[variation.id]?.primary || {};
-    mock.primary = handler(variation.primary, maybeMockConfig);
-
-    const items: Models.VariationMock["items"] = [];
+function buildVariationMockConfig(
+  model: Variation,
+  legacyVariationMockConfig?: Partial<Record<string, Record<string, unknown>>>
+): VariationMockConfig {
+  const primaryFields =
+    model.primary &&
+    buildFieldsMockConfig(
+      model.primary,
+      legacyVariationMockConfig?.["primary"]
+    );
+  const itemFields =
+    model.items &&
+    buildFieldsMockConfig(model.items, legacyVariationMockConfig?.["items"]);
+  return {
+    nbItemsBlocks: 1,
+    primaryFields,
+    itemFields,
+  };
+}
 
-    const repeat = variation.items;
-    if (!repeat || (repeat && repeat.length === 0)) {
+export function buildSliceMockConfig(
+  model: SharedSlice,
+  legacyMockConfig: Partial<
+    Record<string, Partial<Record<string, Partial<Record<string, unknown>>>>>
+  >
+): ReadonlyArray<SharedSliceMockConfig> {
+  const variationConfigs: Record<string, VariationMockConfig> =
+    model.variations.reduce((acc, variationModel) => {
+      const variationMockConfig = legacyMockConfig[variationModel.id];
       return {
-        ...mock,
-        items,
+        ...acc,
+        [variationModel.id]: buildVariationMockConfig(
+          variationModel,
+          variationMockConfig
+        ),
       };
-    }
+    }, {});
 
-    for (let i = 0; i < Math.floor(Math.random() * 6) + 2; i++) {
-      items.push(handler(repeat, mockConfig ? mockConfig.items || {} : {}));
-    }
-
-    mock.items = items;
-    return mock;
+  return model.variations.map((v) => {
+    return {
+      variation: v.id,
+      variations: variationConfigs,
```

Comment:
Should it be
```suggestion
      variations: variationConfigs[v.id],
```
So that each variation is associated to it's personal mock config and the full object is no copied multiple times ?

Also, do you know why is the key called variations ?

Reply:
no you can't, the lib expect a slice to provide a list of config not a single one, it must match the type of a slice not a variation.
The `variation` param before is an optional one to specify which one you want to generate mocks for eventually

Reply:
`the lib expect a slice to provide a list of config`, but each variation can have it's own config, so why would you need the informations for the other variations ?

Reply:
the lib expect a slice definition not a variation one.